### DISCUSSION TITLE
do authorization before filters.

### DISCUSSION THREAD

**Discussion by Boegie19** [directus/directus#20452] [Feature Requests]
### Summary

filters are in the secure enclave (the directus application itself) so they should be able to do more then the user should be allowed to do.

### Basic Example

```ts
export default defineHook(({ action, filter }, {services, getSchema}) => {
	filter('permissions.read', async (input: Record<string, any>) => {
		const value = await ThirdPartyServiceCall()
		input.hiddenFieldTheUserIsNotAllowedToEdit = value

		return input
	});
});
```

### Motivation

To allow me to do a request to a 3rd party service and if the 3rd party service is offline or something else goes wrong it does not Allow for the creation of the record

### Detailed Design

https://github.com/directus/directus/blob/23c286d9c1ef105f0656d4b22d671760d642c9b6/api/src/services/items.ts#L395-L431

Switch everything behind line 414 to line 395 

And do the same for create update and delete 

### Must Have:
- a way to add filters that bypass the authorization checks

### Drawbacks

- end users mind to a change in filters where they expect authorization to be present and the user to get an error. 

### Alternatives

Add a second type of filter that happens after authorization.

### Adoption Strategy

You will move authorization before filters this should not impact anything exists since before it would give 404 errors,

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
